src/redisearch_rs/trie_bencher/src/corpus.rs

```diff
@@ -1,75 +1,200 @@
+        let path = self.get_cached_path();
+        let corpus = if std::fs::exists(&path).ok() != Some(true) {
+            let corpus = download_corpus(self.get_url());
+            // ensure data folder exists
+            let cache_folder = PathBuf::from("data");
+            if !std::fs::exists(cache_folder)
+                .expect("Cannot check for existience of data folder (cache), check permissions")
+            {
+                std::fs::create_dir(PathBuf::from("data"))
+                    .expect("Failed to create data folder (cache)");
+            }
+            fs_err::write(&path, corpus.as_bytes()).expect("Failed to write corpus to disk");
+            corpus
+        } else {
+            fs_err::read_to_string(&path).expect("Failed to read corpus")
+        };
+
+        // check that the corpus hasn't been altered.
+        if let Some(stored_checksum) = self.get_checksum() {
+            let checksum = crc32fast::hash(corpus.as_bytes());
+            assert_eq!(
+                checksum, stored_checksum,
+                "The checksum of the downloaded corpus does not match the expected value. \
+                This may impact the accuracy and relevance of the operations being benchmarked. \
+                Review the diff before updating the expected checksum."
+            );
+        }
         corpus
-    } else {
-        fs_err::read_to_string(&corpus_path).expect("Failed to read corpus")
-    };
-
-    // Build a smaller corpus for microbenchmarks, limited to <100 unique words.
-    let bench_words = {
-        let mut unique = BTreeSet::new();
-        'outer: for line in corpus.lines().skip(36) {
-            for word in line.split_whitespace() {
-                if unique.insert(word.to_string()) && unique.len() > 82 {
-                    break 'outer;
+    }
+
+    /// Creates a vector of keys for insertion into a trie,
+    /// may output the trie to a file using a pretty printer.
+    ///
+    /// The pretty printed version of the trie helps a developer to
+    /// explore the data, i.e. check what node would be inserted at
+    /// what depth.
+    pub fn create_keys(&self, output_pretty_print_trie: bool) -> Vec<String> {
+        let corpus = self.download_or_read_corpus();
+        let reval = match self {
+            CorpusType::RedisBench1kWiki => self.create_keys_redis_wiki1k(&corpus),
+            CorpusType::RedisBench10kNumerics => self.create_keys_redis_wiki10k(&corpus),
+            CorpusType::GutenbergEbook => self.create_keys_gutenberg(&corpus),
+        };
+        if output_pretty_print_trie {
+            let trie = rust_load_from_keys(&reval);
+            fs_err::write(self.get_pretty_print_path(), format!("{trie:?}").as_bytes())
+                .expect("Failed to write bench words debug to disk");
+        }
+        reval
+    }
+
+    fn create_keys_redis_wiki10k(&self, contents: &str) -> Vec<String> {
+        // we find the guid like doc id in column 5
```

Comment:
please mention (here or anywhere you find appropriate) that you use the doc id (document name) as the trie input, as opposed to using values.